packages/relay/test/utils/userHelper.ts

```diff
@@ -0,0 +1,53 @@
+import { Identity } from '@semaphore-protocol/identity'
+import { UserState } from '@unirep/core'
+import crypto from 'crypto'
+import { ethers } from 'hardhat'
+import { UnirepSocialSynchronizer } from '../../src/services/singletons/UnirepSocialSynchronizer'
+import { IdentityObject } from './types'
+
+export function createRandomUserIdentity(): IdentityObject {
+    const hash = crypto.createHash('sha3-224')
+    const hashUserId = `0x${hash
+        .update(new Identity().toString())
+        .digest('hex')}` as string
+    const id = new Identity(hashUserId)
+
+    return { hashUserId, id }
+}
+
+export function createUserIdentities(size: number): IdentityObject[] {
+    let result: IdentityObject[] = []
+    for (let i = 0; i < size; i++) {
+        result.push(createRandomUserIdentity())
+    }
+    return result
+}
+
+export async function genUserState(id, app, db, prover) {
```

Comment:
I don't know if this will occur any conflict while two synchronizers polling the blocks, but even we don't assign a synchronizer, it will be initiated while creating userState. I create synchronizer to make sure the sync will update our custom event when calling `userState.waitForSync()`
https://github.com/Unirep/Unirep/blob/e57834c1244e76a1e3e8289b0f92b072c2fc8fb8/packages/core/src/UserState.ts#L125-L139

Reply:
Can we just pass in the synchronizer?